### PR TITLE
Add early stopping to training

### PR DESCRIPTION
**Pull request by keenanjohnson**

### CODE REVIEW

**Comment by keenanjohnson** [comment]
Adds a new early stopping parameter: `--early-stopping` (default: 15): Stop training if validation loss doesn't improve for N epochs (0 to disable)
